synthesizer/src/process/execute.rs

```diff
@@ -226,4 +220,153 @@ impl<N: Network> Process<N> {
+                // If the stack is empty, then push the `transition` to the top of the stack.
+                None => {
+                    traversal_stack.push(TransitionMetadata::new(
+                        &mut counter,
+                        *transition.program_id(),
+                        *transition.function_name(),
+                        Some(*transition.id()),
+                    ));
+                }
+                // If the stack is not empty, then add the current transition ID to the entry.
+                Some(head) => match head.pid == *transition.program_id() && head.fname == *transition.function_name() {
+                    true => head.tid = Some(*transition.id()),
+                    false => bail!("Unexpected transition in the execution"),
+                },
+            }
+
+            // Process the entry at the top of the stack. By the previous step, this entry has a transition ID.
+            // Note this unwrap is safe, since we either pushed an entry to the stack or modified the one at the top of the stack.
+            let top = traversal_stack.last().unwrap();
+            // If the entry is complete, then add it to the call graph.
+            if top.is_complete() {
+                // Note this unwrap is safe, for the same reason as above.
+                update_call_graph(traversal_stack.pop().unwrap(), &mut call_graph, &mut uid_to_tid)?;
+            } else {
+                // Retrieve the stack.
+                let stack = self.get_stack(top.pid)?;
+                // Retrieve the function from the stack.
+                let function = stack.get_function(&top.fname)?;
+                // Collect the children of the current transition.
+                let children = function
+                    .instructions()
+                    .iter()
+                    .filter_map(|instruction| match instruction {
+                        Instruction::Call(call) => {
+                            let (pid, fname) = match call.operator() {
+                                crate::CallOperator::Locator(locator) => (locator.program_id(), locator.resource()),
+                                crate::CallOperator::Resource(fname) => (&top.pid, fname),
+                            };
+                            Some(TransitionMetadata::new(&mut counter, *pid, *fname, None))
+                        }
+                        _ => None,
+                    })
+                    .collect::<Vec<_>>();
+
+                // Add the children UIDs to the metadata.
+                // Note this unwrap is safe, for the same reason as above.
+                let top = traversal_stack.last_mut().unwrap();
+                let child_uids = children.iter().map(|child| child.uid).collect::<Vec<_>>();
+                match top.children {
+                    None => top.children = Some(child_uids),
+                    Some(ref mut children) => children.extend(child_uids),
```

Comment:
To my knowledge, this `Some(..)` case is never hit. We should replace it with a `Some(..) => bail!("...")`.